ts/ImageList/index.tsx

```diff
@@ -0,0 +1,135 @@
+import autobind from 'autobind-decorator';
+import {h} from 'preact';
+import MaterialComponent from '../Base/MaterialComponent';
+
+export interface IImageListAspectContainerProps {}
+
+export interface IImageListAspectContainerState {}
+
+export class ImageListAspectContainer extends MaterialComponent<
+  IImageListAspectContainerProps,
+  IImageListAspectContainerState
+> {
+  protected componentName = 'image-list__image-aspect-container';
+  protected mdcProps = [];
+
+  @autobind
+  protected materialDom() {
+    return <div>{this.props.children}</div>;
+  }
+}
+
+export interface IImageListItemProps {}
+
+export interface IImageListItemState {}
+
+export class ImageListItem extends MaterialComponent<
+  IImageListItemProps,
+  IImageListItemState
+> {
+  protected componentName = 'image-list__item';
+  protected mdcProps = [];
+
+  @autobind
+  protected materialDom(props) {
+    return (
+      <li {...props} ref={this.setControlRef}>
+        {props.children}
+      </li>
+    );
+  }
+}
+
+export interface IImageListImageProps {
+  src: string;
+}
+
+export interface IImageListImageState {}
+
+export class ImageListImage extends MaterialComponent<
+  IImageListImageProps,
+  IImageListImageState
+> {
+  protected componentName = 'image-list__image';
+  protected mdcProps = [];
+
+  @autobind
+  protected materialDom() {
+    const {src, ...props} = this.props;
+    return (
+      <img src={src} {...props} ref={this.setControlRef}>
```

Comment:
Why did you separate `src` from `props`? As I see it it should be unnecessary to set `src` independent of `props`.

Reply:
Because my ide complained that src is necessary on <img> tags

Reply:
aah okay. That's fine I guess.